exercises/reverse-string/canonical-schema.json

```diff
@@ -0,0 +1,54 @@
+{
+  "exercise": "reverse-string",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "empty string",
+      "property": "reverse-string",
+      "input": "",
+      "expected": ""
+    },
+       {
+      "description": "reverse a single character string",
+      "property": "reverse-string",
+      "input": "e",
+      "expected": "e"
+    },
+    {
+      "description": "reverse a string",
+      "property": "reverse-string",
+      "input": "robot",
+      "expected": "tobor"
+    },
+    {
+      "description": "reverse a string consisting of a single lowercase letter",
```

Comment:
Description does not match test.

Reply:
a “string consisting of a single lowercase letter” has a length of 1, your string has a length of 4.

Reply:
Oh I see, I read the test as any string length as long as string as 1 single lowercase letter, not a string that ONLY has a single lowercase letter. If the only single letter case is more clear, then I can just update that.

Reply:
I don't think character case is something that needs to be tested in this exercise. Seems a superfluous detail.

Reply:
As I've explained in another comment, I've seen a lot of code that has arbitrary “validation” built in, because there were no examples with uppercase values…

Therefore I think, we should explicitely show, that **any** kind of input is to be expected.

Reply:
I think it's OK to have some mixed case examples. I also think that this test suite spends too much time concentrating on case based tests.

A capitalised palindrome might be an interesting test.